inc/functions/recipient.php

```diff
@@ -6,44 +6,39 @@
  */
 
 use BracketSpace\Notification\Interfaces;
+use BracketSpace\Notification\Defaults\Store\Recipient as RecipientStore;
 
 /**
  * Registers recipient
  * Uses notification/recipients filter
  *
- * @since  6.0.0 Changed naming convention from Notification to Carrier.
+ * @since  [Next] Added Recipient Store
  * @param  string                $carrier_slug Carrier slug.
  * @param  Interfaces\Receivable $recipient    Recipient object.
- * @return void
+ * @return \WP_Error | true
  */
 function notification_register_recipient( $carrier_slug, Interfaces\Receivable $recipient ) {
+	$store = new RecipientStore();
 
-	add_filter( 'notification/recipients', function( $recipients ) use ( $carrier_slug, $recipient ) {
-
-		if ( ! isset( $recipients[ $carrier_slug ] ) ) {
-			$recipients[ $carrier_slug ] = [];
-		}
-
-		if ( isset( $recipients[ $carrier_slug ][ $recipient->get_slug() ] ) ) {
-			throw new \Exception( 'Recipient with that slug is already registered for ' . $carrier_slug . ' Carrier' );
-		} else {
-			$recipients[ $carrier_slug ][ $recipient->get_slug() ] = $recipient;
-		}
-
-		return $recipients;
-
-	} );
+	try {
+		$store[ $carrier_slug ] = $recipient;
```

Comment:
You should group the Recipient with Carrier slug, like this:

```suggestion
		$store[ $carrier_slug ][ $recipient->get_slug() ] = $recipient;
```

So in the `notification_get_carrier_recipients` function you'll be able to just pick up the first key and get all Recipients for particular Carrier slug.

Reply:
The above will allow the plugin to share the same recipient for multiple carriers, like this:

```
'email' => Array(
    'email_recipient' => EmailRecipient,
    'user_recipient'  => UserRecipient,
),
'sendgrid' => Array(
    'email_recipient' => EmailRecipient,
)
```